src/Event.cpp

```diff
@@ -6,35 +7,54 @@
 #include "HGCalSimulation/FastShower/interface/Event.h"
 #endif
 
+Event::Event(uint32_t run, uint32_t event): run_(run), event_(event) {
+}
+
+void Event::fillHit(uint32_t id, double energy) {
+    auto itr_insert = hits_.emplace(id, energy);
+    // if id already inserted (noise and hit), add the energy
+    if(!itr_insert.second) itr_insert.first->second += energy;
+}
+
+void Event::fillGenEn(uint32_t id, double energy) {
+    generated_energy_.emplace(id, energy);
+}
+
+void Event::fillGenEta(uint32_t id, double eta) {
+    generated_eta_.emplace(id, eta);
+}
+
+void Event::fillGenPhi(uint32_t id, double phi) {
+    generated_phi_.emplace(id, phi);
+}
+
+void Event::fillPDGid(uint32_t id, int PDGid) {
+    pdgid_.emplace(id, PDGid);
+}
+
+void Event::fillThick(uint32_t id, int thick) {
+    if (!(thick_.find(id) != thick_.end()))
+        thick_.emplace(id, thick);
+}
 
-Event::
-Event(uint32_t run, uint32_t event):
-  run_(run),
-  event_(event)
-{
+void Event::fillCells(uint32_t id, const Cell& cell) {
+    // if the cell is already in map, don't add it
+    if (!(cells_.find(id) != cells_.end()))
+        cells_.emplace(id, cell);
 }
 
-void
-Event::
-fillHit(uint32_t id, double energy)
-{
-  auto itr_insert = hits_.emplace(id, energy);
-  // if id already inserted, add the energy
-  if(!itr_insert.second) itr_insert.first->second += energy;
+void Event::setnPart(uint32_t part) {
+    npart_ = part;
 }
 
-void
-Event::
-setGenerated(double energy, double eta, double phi)
-{
-  generated_energy_ = energy;
-  generated_eta_ = eta;
-  generated_phi_ = phi;
+void Event::clear() {
+    hits_.clear();
 }
 
-void
-Event::
-clear()
-{
-  hits_.clear();
+
+int Event::getLayerFromId(int cell_id) {
+
+    int len = std::to_string(cell_id).length();
```

Comment:
I don't understand at all this string manipulation. You should only do bitwise operations.

Reply:
This string manipulation allows to access layer id with the cell id. As the cell id is iii0jjj0kk with k the layer id you need to take the last to number to access the layer. Maybe there is a way to do this with bit operation ?

Reply:
This is really not the standard way to do this, it is really complicated and string manipulations are costly. Please use bits manipulation.